// @tanstack/ai/dist/esm/activities/generateVideo/index.js
const kind = "video";
async function generateVideo(options) {
  const { adapter, prompt, size, duration, modelOptions } = options;
  const model = adapter.model;
  return adapter.createVideoJob({
    model,
    prompt,
    size,
    duration,
    modelOptions
  });
}
async function getVideoJobStatus(options) {
  const { adapter, jobId } = options;
  const statusResult = await adapter.getVideoStatus(jobId);
  if (statusResult.status === "completed") {
    try {
      const urlResult = await adapter.getVideoUrl(jobId);
      return {
        status: statusResult.status,
        progress: statusResult.progress,
        url: urlResult.url
      };
    } catch (error) {
      return {
        status: statusResult.status,
        progress: statusResult.progress,
        error: error instanceof Error ? error.message : "Failed to get video URL"
      };
    }
  }
  return {
    status: statusResult.status,
    progress: statusResult.progress,
    error: statusResult.error
  };
}
function createVideoOptions(options) {
  return options;
}
export {
  createVideoOptions,
  generateVideo,
  getVideoJobStatus,
  kind
};
