// @tanstack/ai/dist/esm/index.js
import { chat, createChatOptions } from "./activities/chat/index.js";
import { createSummarizeOptions, summarize } from "./activities/summarize/index.js";
import { createImageOptions, generateImage } from "./activities/generateImage/index.js";
import { createVideoOptions, generateVideo, getVideoJobStatus } from "./activities/generateVideo/index.js";
import { createSpeechOptions, generateSpeech } from "./activities/generateSpeech/index.js";
import { createTranscriptionOptions, generateTranscription } from "./activities/generateTranscription/index.js";
import { toolDefinition } from "./activities/chat/tools/tool-definition.js";
import { convertSchemaToJsonSchema } from "./activities/chat/tools/schema-converter.js";
import { streamToText, toHttpResponse, toHttpStream, toServerSentEventsResponse, toServerSentEventsStream } from "./stream-to-response.js";
import { ToolCallManager } from "./activities/chat/tools/tool-calls.js";
import { combineStrategies, maxIterations, untilFinishReason } from "./activities/chat/agent-loop-strategies.js";
import { aiEventClient } from "./event-client.js";
import { convertMessagesToModelMessages, generateMessageId, modelMessageToUIMessage, modelMessagesToUIMessages, normalizeToUIMessage, uiMessageToModelMessages } from "./activities/chat/messages.js";
import { StreamProcessor, createReplayStream } from "./activities/chat/stream/processor.js";
import { BatchStrategy, CompositeStrategy, ImmediateStrategy, PunctuationStrategy, WordBoundaryStrategy } from "./activities/chat/stream/strategies.js";
import { PartialJSONParser, defaultJSONParser, parsePartialJSON } from "./activities/chat/stream/json-parser.js";
export {
  BatchStrategy,
  CompositeStrategy,
  ImmediateStrategy,
  PartialJSONParser,
  PunctuationStrategy,
  StreamProcessor,
  ToolCallManager,
  WordBoundaryStrategy,
  aiEventClient,
  chat,
  combineStrategies,
  convertMessagesToModelMessages,
  convertSchemaToJsonSchema,
  createChatOptions,
  createImageOptions,
  createReplayStream,
  createSpeechOptions,
  createSummarizeOptions,
  createTranscriptionOptions,
  createVideoOptions,
  defaultJSONParser,
  generateImage,
  generateMessageId,
  generateSpeech,
  generateTranscription,
  generateVideo,
  getVideoJobStatus,
  maxIterations,
  modelMessageToUIMessage,
  modelMessagesToUIMessages,
  normalizeToUIMessage,
  parsePartialJSON,
  streamToText,
  summarize,
  toHttpResponse,
  toHttpStream,
  toServerSentEventsResponse,
  toServerSentEventsStream,
  toolDefinition,
  uiMessageToModelMessages,
  untilFinishReason
};
